### PR TITLE
A few changes - priority class, k8s 1.17 and extra options

### PR DESCRIPTION
**Pull request by garthwilliamson**
Hi,

I have a few commits that we found useful. I'm happy for you to cherry pick these as needed, or I can make seperate pull requests or make any changes you'd prefer.

I don't know how compatible the addional rules for the cluster role is with older kubernetes versions.

Cheers,
Garth

### CODE REVIEW

File: aws.tf [hunk 1]
```diff
@@ -27,7 +27,7 @@ EOF
 }
 
 resource "aws_iam_role_policy_attachment" "cluster_autoscaler" {
-  count = var.aws_create_iam_policy && var.aws_iam_role_for_policy != null ? 1 : 0
+  count = var.aws_create_iam_policy ? 1 : 0
```
Comment (MartinSadovy) [inline aws.tf]: Then make aws_iam_role_for_policy mandatory.
And separate it to the second commit because it is BC.

File: variables.tf [hunk 2]
```diff
@@ -73,3 +73,9 @@ variable "aws_iam_role_for_policy" {
   default = null
   description = "AWS IAM Role name for attaching AWS IAM policy."
 }
+
+variable "additional_autoscaler_options" {
+  type        = list(string)
+  default     = []
+  description = "Additional options to pass to the autoscaler."
```
Comment (MartinSadovy) [inline variables.tf]:
```suggestion
  description = "Additional options to pass to the autoscaler command."
```

File: aws.tf (hunk 1 above)
Comment (garthwilliamson) [reply aws.tf]: I was getting strange behaviour without this change, however my fix here is obviously wrong. I'm going to update this pull request without this change in it, and when I next do a full build I'll work out what I was actually trying to fix.

**Comment by garthwilliamson** [comment]
I've updated this pull request without the commit to remove the check on check aws_iam_role_for_policy. I'll create a new pull request for that one once I understand exactly why the error was occuring and if it's avoidable in a different way.

**Comment by Vrtak-CZ** [comment]
merged by individual commits

**Pull request closed without merging**